### PR TITLE
feat: [SRM-14145]: Getting rid of two enums for connectors

### PR DESCRIPTION
**Pull request by harshilbits**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.utils.ts [hunk 1]
```diff
@@ -15,17 +17,13 @@ export function shouldShowChartComponent(
...
-  // query: string
 ): boolean {
-  // return !!(query && chartConfig?.enabled && records && records?.length && !fetchingSampleRecordLoading)
   return Boolean(chartConfig?.enabled && !(isQueryRuntimeOrExpression || isConnectorRuntimeOrExpression))
 }
 
 export function getRecordsRequestBody(
   connectorIdentifier: any,
-  providerType: string,
+  healthSourceType: string | undefined,
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.utils.ts:26]: please use optional operator `healthSourceType?: string`

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/__tests__/CustomMetricFormContainer.mock.ts [hunk 2]
```diff
@@ -152,7 +152,7 @@ export const logsTablePayloadMock = {
   connectorIdentifier: 'Sumo_logic',
   endTime: expect.any(Number),
   healthSourceQueryParams: { serviceInstanceField: '_sourcehost' },
-  providerType: 'SUMOLOGIC_LOG',
+  healthSourceType: 'SumologicLogs',
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/__tests__/CustomMetricFormContainer.mock.ts:155]: NIT: should we pick from constant HealthSourceTypes.SumologicLogs

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/__tests__/CustomMetricFormContainer.test.tsx [hunk 3]
```diff
@@ -488,7 +488,7 @@ describe('Unit tests for CustomMetricFormContainer', () => {
         connectorIdentifier: 'Sumo_logic',
         endTime: expect.any(Number),
         healthSourceQueryParams: { serviceInstanceField: '_sourcehost' },
-        providerType: 'SUMOLOGIC_LOG',
+        healthSourceType: 'SumologicLogs',
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/__tests__/CustomMetricFormContainer.test.tsx:491]: NIT: should we pick from constant HealthSourceTypes.SumologicLogs

**Review by deepesh-ui — APPROVED** [review]

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.utils.ts (hunk 1 above)
Comment (harshilbits) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.utils.ts:26]: Isn't it same ?

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/__tests__/CustomMetricFormContainer.test.tsx (hunk 3 above)
Comment (harshilbits) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/__tests__/CustomMetricFormContainer.test.tsx:491]: same.

**Review by praneshg239 — APPROVED** [review]

**Pull request merged**